src/context.cc

```diff
@@ -1543,6 +1528,22 @@ const std::string Context::shortOSName() {
     return "unknown";
 }
 
+void Context::parseVersion(int result[4], const std::string& input) {
+    std::istringstream parser(input);
+    parser >> result[0];
+    for (auto idx = 1; idx < 4; idx++) {
+        parser.get(); //Skip period
+        parser >> result[idx];
+    }
+}
+
+bool Context::lessThanVersion(const std::string& version1, const std::string& version2) {
+    int parsed1[4] {}, parsed2[4] {};
+    parseVersion(parsed1, version1);
+    parseVersion(parsed2, version2);
+    return std::lexicographical_compare(parsed1, parsed1 + 4, parsed2, parsed2 + 4);
```

Comment:
Personally, I find this notion a tad confusing and very C-like. This may be just me though, I'd personally write `&parsed1[4]`, it seems a lot clearer to me.

Reply:
👌